Content.Server/Fluids/EntitySystems/AbsorbentSystem.cs

```diff
@@ -117,30 +117,43 @@ private bool TryTransferAbsorber(EntityUid user, EntityUid used, EntityUid targe
         if (!_solutionSystem.TryGetRefillableSolution(target, out var refillableSolution, refillable: refillable))
             return false;
 
-        if (refillableSolution.Volume <= 0)
+        if (refillableSolution.Volume < 0)
         {
             var msg = Loc.GetString("mopping-system-target-container-empty", ("target", target));
             _popups.PopupEntity(msg, user, user);
             return false;
         }
 
+        var transferAmount = component.PickupAmount / PuddleSystem.EvaporationReagentRatio -
+            absorberSoln.GetReagentQuantity(PuddleSystem.EvaporationReagent);
```

Comment:
I'll start commenting here in order to explain the code, maybe that helps in getting it merged:

These two lines add a new limit to the amount the mop will pickup: Instead of adding up to PickupAmount of water, it will only add a fraction of it, the same fraction which is needed to be able to clean a puddle of PickupAmount volume.
A subtraction makes it so that only the remaining water needed to reach PickupAmount / Ratio is removed from the container.

Reply:
If you find yourself doing this in a PR, consider if it would be better to write this explanation as comments in the code.

Reply:
Can I? If I did that maybe it would bee too long to review. I don't really know. There's plenty of comments in the code already, including some addressing questions being asked here, same goes for the comments.